event-hubs.api.md

```diff
@@ -0,0 +1,275 @@
+import { WebSocketImpl } from 'rhea-promise';
+
+// @public
+export interface Checkpoint extends PartitionContext {
+    offset: number;
+    sequenceNumber: number;
+}
+
+// @public
+export enum CloseReason {
+    OwnershipLost = "OwnershipLost",
+    Shutdown = "Shutdown"
+}
+
+// @public
+export interface CreateBatchOptions {
+    abortSignal?: AbortSignalLike;
+    maxSizeInBytes?: number;
+    partitionId?: string;
+    partitionKey?: string;
+}
+
+// @public
+export interface EventData {
+    body: any;
+    properties?: {
+        [key: string]: any;
+    };
+}
+
+// @public
+export interface EventDataBatch {
+    // @internal
+    batchMessage: Buffer | undefined;
+    count: number;
+    // @internal
+    _messageSpanContexts: SpanContext[];
+    partitionId?: string;
+    partitionKey?: string;
+    sizeInBytes: number;
+    tryAdd(eventData: EventData, options?: TryAddOptions): boolean;
+}
+
+// @public
+export interface EventHubClientOptions {
+    retryOptions?: RetryOptions;
+    userAgent?: string;
+    webSocket?: WebSocketImpl;
+    webSocketConstructorOptions?: any;
+}
+
+// @public
+export class EventHubConsumerClient {
+    constructor(connectionString: string, options?: EventHubClientOptions);
+    constructor(connectionString: string, eventHubName: string, options?: EventHubClientOptions);
+    constructor(fullyQualifiedNamespace: string, eventHubName: string, credential: TokenCredential, options?: EventHubClientOptions);
+    close(): Promise<void>;
+    static defaultConsumerGroupName: string;
+    getEventHubProperties(options?: GetEventHubPropertiesOptions): Promise<EventHubProperties>;
+    getPartitionIds(option?: GetPartitionIdsOptions): Promise<string[]>;
+    getPartitionProperties(partitionId: string, options?: GetPartitionPropertiesOptions): Promise<PartitionProperties>;
+    subscribe(consumerGroupName: string, handlers: SubscriptionEventHandlers, options?: SubscriptionOptions): Subscription;
+    subscribe(consumerGroupName: string, partitionId: string, handlers: SubscriptionEventHandlers, options?: SubscriptionOptions): Subscription;
+    subscribe(consumerGroupName: string, partitionManager: PartitionManager, handlers: SubscriptionEventHandlers, options?: SubscriptionOptions): Subscription;
+}
+
+// @public
+export class EventHubProducerClient {
+    constructor(connectionString: string, options?: EventHubClientOptions);
+    constructor(connectionString: string, eventHubName: string, options?: EventHubClientOptions);
+    constructor(fullyQualifiedNamespace: string, eventHubName: string, credential: TokenCredential, options?: EventHubClientOptions);
+    close(): Promise<void>;
+    createBatch(options?: CreateBatchOptions): Promise<EventDataBatch>;
+    readonly eventHubName: string;
+    readonly fullyQualifiedNamespace: string;
+    getEventHubProperties(options?: GetEventHubPropertiesOptions): Promise<EventHubProperties>;
+    getPartitionIds(options?: GetPartitionIdsOptions): Promise<Array<string>>;
```

Comment:
This could probably be named `listPartitionIds` since this is returning an array/list of PartitionIds rather than a single partitionId the [client naming conventions](https://azure.github.io/azure-sdk/typescript_design.html) talks a bit about this. What do you think?